### PR TITLE
PWX-37570 | Add missing fields for updateStrategy

### PR DESCRIPTION
**Pull request by hitesh-wani-px**
**What this PR does / why we need it**:
Added missing fields for updateStrategy component in storageCluster template

**Which issue(s) this PR fixes** (optional)
Closes #https://purestorage.atlassian.net/browse/PWX-37570

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by hitesh-wani-px** [comment]
**Testing Notes:** 
1. RollingUpdate
   1. Enter all possible values for updateStrategy RollingUpdate in `values.yaml`
        (Enter old PX version and then upgrade to latest)
    ```yaml
    imageVersion: 3.0.5
    updateStrategy:   
      type:    RollingUpdate                             
      maxUnavailable:  3                     
      minReadySeconds: 30                     
      autoUpdateComponents: Always                 
    internalKVDB: true
    ```
   2. Install PX
    ```
    helm install my-release --set internalKVDB=true,clusterName=$(uuidgen) ./charts/portworx/ -n portworx --debug
    ```
    3. The generated storageCluster template should incorporate all specified values.
     ```yaml
    kind: StorageCluster
    apiVersion: core.libopenstorage.org/v1
    metadata:
      name: "mycluster"
      namespace: kube-system
      annotations:
      labels:
        heritage: "Helm"
        release: "my-release"
        chart: "portworx-3.1.0"
        app.kubernetes.io/managed-by: "Helm"
        app.kubernetes.io/instance: "my-release"
    spec:
      image: portworx/oci-monitor:3.0.5
      imagePullPolicy: Always
      updateStrategy:
        type: RollingUpdate
        rollingUpdate:
          maxUnavailable: 3
          minReadySeconds: 30
      autoUpdateComponents: Always

       kvdb:
        internal: true
      storage:
        useAll: true
      secretsProvider: k8s

      env:
      - name: PX_SECRETS_NAMESPACE
        value: portworx

      stork:
        enabled: true
      csi:
        enabled: false
     ```
     4. After successfully installation of PX upgrade PX version to 3.1.0
     ```
     helm upgrade my-release --set internalKVDB=true,imageVersion=3.1.0 ./charts/portworx/ -n portworx
     ```
     During upgrade max 3 nodes should be down at time for performing rolling update 
     and upgraded pods to be ready for at least minReadySeconds before updating the next batch of pods.
2. OnDelete
   1. Add upgrade strategy type to `OnDelete`
        (Enter old PX version and then upgrade to latest)
    ```yaml
        imageVersion: 3.0.5
        updateStrategy:   
          type:    OnDelete                             
        autoUpdateComponents: Always                 
        internalKVDB: true
    ```
    2. Install PX
    ```
    helm install my-release --set internalKVDB=true,clusterName=$(uuidgen) ./charts/portworx/ -n portworx --debug
    ```
    3. The generated storageCluster yaml should have upgradeStrategy as `OnDelete`
    4. After successfully installation of PX upgrade PX version to 3.1.0
     ```
     helm upgrade my-release --set internalKVDB=true,imageVersion=3.1.0 ./charts/portworx/ -n portworx
     ```
    5. Delete one of the pod manually to perform upgrade for `OnDelete` strategy. After that we should be able to see the new px 
        pod with upgraded version (3.1.0).
        ```
        image: docker.io/portworx/oci-monitor:3.1.0
        ```

File: charts/portworx/values.yaml [hunk 1]
```diff
@@ -115,3 +115,14 @@ clusterToken:
 
 deleteStrategy:                           # Optional: Delete strategy for the portworx cluster
   type:                                   # Valid values: Uninstall, UninstallAndWipe
+
+updateStrategy:   
+  type:                                   # portworx supports the following update strategies- RollingUpdate and OnDelete
```
Comment (nikita-bhatia) [inline charts/portworx/values.yaml:120]: should we add RollingUpdate as default?

Comment (hitesh-wani-px) [reply charts/portworx/values.yaml:120]: We are only adding the updateStrategy in storageCluster if the user has specified valid values. The operator can handle the default values later if we not specifies any value in type.

**Comment by nikita-bhatia** [comment]
PR looks good to me.
I would want @vinayakshnd @diptiranjanpx to confirm if we want to set default value of UpdateStrategy as mentioned in [PWX-29986](https://github.com/portworx/helm/tree/PWX-29986)
Since from portworx-central, by default it is not added, I am approving the PR.

**Review by nikita-bhatia — APPROVED** [review]

**Pull request merged**